### DISCUSSION TITLE
Deprecating UPN based APIs

### DISCUSSION THREAD

**Discussion by boyuzhang-msft** [microsoftconnect/ms-intune-app-sdk-android#238] [General]
## Deprecating UPN based APIs

MAM SDK API methods that take UPNs to specify identities are deprecated since MAM SDK `10.2.1`. They will be removed completely at the next major version increment. New methods that specify identities by OID (also known as AAD User ID, AAD ID or Entra ID) should be used instead. See the [MAM SDK integration guide] and [javadocs] for details.

## Why should apps not use UPN

UPNs are not suitable to uniquely identify identities because they are not only mutable, but different
identities can have the same UPN in guest tenant scenarios.
> **Note:** In the guest tenant scenario, an existing AAD account is added as a guest to another tenant. The
> guest account has a different OID, but can have the same UPN as the associated home tenant account.

## How will this affect your app

Apps are strongly encouraged to call into OID-based API methods. [MAM Strict Mode] checks will fail if calling into UPN-based APIs are detected. Starting from the next major version increment (i.e. SDK version `11.x.x`), apps will no longer be able to call the deprecated UPN-based APIs.

## What do you need to do to migrate

Our [javadocs] mark existing UPN-based API methods as deprecated and link the new OID-based API methods to use. Apps are strongly encouraged to examine their existing usage of UPN-based APIs and migrate to the corresponding OID-based APIs. To learn more on how to obtain OID, please refer to our doc on [MAMEnrollmentManager and Registration].

[javadocs]: https://msintuneappsdk.github.io/ms-intune-app-sdk-android/reference/classes.html
[MAM SDK integration guide]: https://learn.microsoft.com/en-us/mem/intune/developer/app-sdk-android-phase1
[MAM Strict Mode]: https://learn.microsoft.com/en-us/mem/intune/developer/app-sdk-android-phase4#mam-strict-mode
[MAMEnrollmentManager and Registration]: https://learn.microsoft.com/en-us/mem/intune/developer/app-sdk-android-phase4#mamenrollmentmanager-and-registration

**Comment by richakhanna** [comment]
@boyuzhang-msft could you correct the javadocs link? Currently it links to a 404 page.

From:
https://msintuneappsdk.github.io/ms-intune-app-sdk-android/reference/classes.html

To:
https://microsoftconnect.github.io/ms-intune-app-sdk-android/reference/packages.html
